ws-nextjs-app/pages/[service]/send/[id]/FormField/index.tsx

```diff
@@ -15,78 +22,98 @@ const Label = ({
   </Text>
 );
 
-const TextInput = ({ id, name }: InputProps) => {
-  const { handleChange, formState } = useFormContext();
+const InvalidMessageBox = ({ id, messageCode }: InvalidMessageBoxProps) => {
+  const { lang } = useContext(ServiceContext);
+  const message =
+    formTranslations[lang][messageCode ?? InvalidMessageCodes.FieldRequired];
+
+  return (
+    <div>
+      <p id={id}>
+        <span>❗</span>
+        {message}
+      </p>
+    </div>
+  );
+};
 
+const TextInput = ({ id, name, handleChange, inputState }: InputProps) => {
+  const { invalid, value = '', required } = inputState;
   return (
     <input
       id={id}
       name={name}
       type="text"
-      value={(formState?.[name] as string) ?? ''}
+      value={value as string}
       onChange={e => handleChange(e.target.name, e.target.value)}
+      aria-invalid={invalid}
+      aria-required={required}
     />
   );
 };
 
-const TextArea = ({ id, name }: InputProps) => {
-  const { handleChange, formState } = useFormContext();
-
+const TextArea = ({ id, name, handleChange, inputState }: InputProps) => {
+  const { invalid, value = '', required } = inputState;
   return (
     <textarea
       id={id}
       name={name}
-      value={(formState?.[name] as string) ?? ''}
+      value={value as string}
       onChange={e => handleChange(e.target.name, e.target.value)}
+      aria-invalid={invalid}
+      aria-required={required}
     />
   );
 };
 
-const EmailInput = ({ id, name }: InputProps) => {
-  const { handleChange, formState } = useFormContext();
-
+const EmailInput = ({ id, name, handleChange, inputState }: InputProps) => {
+  const { invalid, value = '', required } = inputState;
   return (
     <input
       id={id}
       name={name}
       type="email"
-      value={(formState?.[name] as string) ?? ''}
+      value={value as string}
       onChange={e => handleChange(e.target.name, e.target.value)}
+      aria-invalid={invalid}
+      aria-required={required}
+      formNoValidate
     />
   );
 };
 
-const Checkbox = ({ id, name }: InputProps) => {
-  const { handleChange, formState } = useFormContext();
-
+const Checkbox = ({ id, name, handleChange, inputState }: InputProps) => {
+  const { invalid, value = false, required } = inputState;
   return (
     <input
       id={id}
       name={name}
       type="checkbox"
-      checked={(formState?.[name] as boolean) ?? false}
+      checked={value as boolean}
       onChange={e => handleChange(e.target.name, e.target.checked)}
+      aria-invalid={invalid}
+      aria-required={required}
```

Comment:
Should we use `required` here instead of `aria-required`? `required` is a native input feature and will automatically give us validation without any extra JS.

Reply:
Yeah, I did initially, but the UX deviates quite a bit to what web-core have, I've basically just copied what they have.

Reply:
Ah interesting. May be worth checking with Catharine to see the requirements. I just noticed in the MDN docs that using only `aria-required` doesn't actually make the field required: https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/required

Reply:
After chatting with @shayneahchoon the approach taken is correct as we'll be manually validating the fields to allow for custom UI treatment if validation fails.